### PR TITLE
add gh action to publish package to pypi

### PR DESCRIPTION
**Pull request by camallen**
Related to https://github.com/mwalmsley/zoobot/issues/18#issuecomment-1278635788 

This PR adds an auto CI release mechanism for publishing zoobot to pypi. It uses the GH action to release to pypi https://github.com/pypa/gh-action-pypi-publish

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
```
Comment (camallen) [inline .github/workflows/python-publish.yml:34]: I'm not sure what cmds we'd need here to build and publish the package correctly.

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (camallen) [inline .github/workflows/python-publish.yml:39]: We will need a token added to repo secrets to allow this action to publish the repo to pypi

**Comment by camallen** [comment]
Here is an example publish GH actions workflow from Coleman https://github.com/zooniverse/aggregation-for-caesar/blob/12baac1e1dc0b06022dd93d21942caefce173617/.github/workflows/publish-to-pypi.yml#L15-L25

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (mwalmsley) [reply .github/workflows/python-publish.yml:34]: That's all I think, at least for my local manual releases - nothing fancy

File: .github/workflows/python-publish.yml (hunk 2 above)
Comment (mwalmsley) [reply .github/workflows/python-publish.yml:39]: I think I have added this - security/actions/PYPI_API_TOKEN

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (mwalmsley) [reply .github/workflows/python-publish.yml:34]: I checked. My build steps are:

python -m pip install --upgrade setuptools wheel twine
python setup.py sdist bdist_wheel
python -m twine upload dist/*

Comment (mwalmsley) [reply .github/workflows/python-publish.yml]: Updated

**Comment by mwalmsley** [comment]
Happy with this once testable :) WIll let you carry on with the other actions PR first

**Comment by camallen** [comment]
I've added a `workflow_dispatch` button which should allow you to use the https://github.com/mwalmsley/zoobot/actions/workflows/python-publish.yml UI as well

**Review by mwalmsley — APPROVED** [review]
LGTM - thank you Cam, very helpful QOL change. Will test in anger shortly

**Pull request merged**